crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -572,9 +608,16 @@ mod tests {
                     predicate::eq(true),
                 )
                 .return_once(move |_, _, _, _| Ok(vec![gas_coin_clone]));
+            let metrics = KeyValueStoreMetrics::new_for_tests();
+            let transaction_kv_store = Arc::new(TransactionKeyValueStore::new(
+                "rocksdb",
+                metrics,
+                self.clone(),
+            ));
```

Comment:
consider making a CoinReadInternalImpl::new_for_testing method to remove this duplication maybe?